code/modules/research/xenobiology/xenobio_camera.dm

```diff
@@ -199,9 +209,12 @@
 			var/mob/living/carbon/monkey/food = new /mob/living/carbon/monkey(remote_eye.loc, TRUE, owner)
 			if (!QDELETED(food))
 				food.LAssailant = C
-				X.monkeys --
-				to_chat(owner, "[X] now has [X.monkeys] monkeys left.")
-	else
+				X.monkeys--
+				X.monkeys = round(X.monkeys, 0.1)		//Prevents rounding errors
```

Comment:
Will this ever not be an integer?

Previously eating monkeys gave the console itself +0.2 monkeys, but now you HAVE to use the recycler for cubes which will always give +1 monkey when inserted into a console

Reply:
You seem to be under the misconception that we add the monkeys to the recycler though the console and grind them, and that isn't really what's happening. This is just using the parts from the recycler to upgrade the recycling through the console as well as the manual grinding. The increase in monkeys available happens just like before (though higher numbers with upgrades), and will be a decimal number most of the time. 

I believe we discussed that point in a previous review, although it was when we picked them up. Due to giving chat output directly from this variable, it needs to round or you'll get imprecise numbers output to chat. Now we might be okay only rounding on pickup, but if I am being honest, it's been almost a month so I no longer remember precisely what I tested to fix the rounding errors.

Reply:
Currently:
- Suck up monkey via console
- Console gets 0.2
- Recycler gets 0

- Dispense Monkey via console
- Console loses 1
- Recycler loses 0 (not including cube dispensed from recycler)

PR:
- Suck up monkey via console
- Console gets 0
- Recycler gets (math in tenths)

- Dispense Monkey via console
- Console loses 1
- Recycler loses 0 (not including cube dispensed from recycler)

The ONLY way to get monkeys via console now is to use cubes which ONLY give pure whole numbers.

Basically i'm suggesting turning the second arg into 1 or dropping it. I have no issue with the round thing since you explained that already. Does that make sense lol?

Reply:
You still got it wrong. There are two things happening in this PR, not one. The monkey recycler has had the parts changed to do different stuff than before, to give the new ratio of 0.2-0.8 and 
 to be equal to using the consoles recycle action.
The console uses the production var from the recycler to check how many monkeys are gained when pulling up a monkey, and adds this to the consoles monkeys var.

At no point does the console add stored_matter to the recycler, or vice versa.

So sucking up a monkey after the PR:
- Suck up monkey via console
- Console gets 0.2-0.8 monkeys
- Recycler gets 0

- Dispense monkey via console
- Console loses 1

In short, the monkeys are added directly to the console and never really visits the recycler unless you do it manually.

Reply:
correct, this is why you don't second guess your first review 8)